### PR TITLE
Inflate BTF FuncProto params

### PR DESCRIPTION
**Pull request by brycekahle**
In support of #114

### CODE REVIEW

File: internal/btf/types.go [hunk 1]
```diff
@@ -528,7 +535,20 @@ func inflateRawTypes(rawTypes []rawType, rawStrings stringTable) (namedTypes map
 			typ = fn
 
 		case kindFuncProto:
-			fp := &FuncProto{id, nil}
+			rawparams := raw.data.([]btfParam)
+			params := make([]FuncParam, 0, len(rawparams))
+			for _, param := range rawparams {
+				// OK for func params to not have a name
+				name, _ := rawStrings.LookupName(param.NameOff)
```
Comment (lmb) [inline internal/btf/types.go:542]: From `Documentation/bpf/btf.rst`:
```rst
If a BTF_KIND_FUNC_PROTO type is referred by a BTF_KIND_FUNC type, then
``btf_param.name_off`` must point to a valid C identifier except for the
possible last argument representing the variable argument. The btf_param.type
refers to parameter type.

If the function has variable arguments, the last parameter is encoded with
``name_off = 0`` and ``type = 0``.
```

We should only try to resolve a name if NameOff != 0. From the description it sounds a bit like the rule doesn't apply if FUNC_PROTO is not referred to by a FUNC, can you tell me more about this?

**Review by lmb — CHANGES_REQUESTED** [review]

File: internal/btf/types.go (hunk 1 above)
Comment (brycekahle) [reply internal/btf/types.go:542]: From the tests, it looks like attempting to resolve `0` is actually OK, and results in `""`. I can reinstate the error handling. https://github.com/cilium/ebpf/blob/4048cd641690f2f82d57b130467a3fdfe4967a06/internal/btf/strings_test.go#L25

Comment (brycekahle) [reply internal/btf/types.go]: Updated.

Comment (lmb) [reply internal/btf/types.go]: Yes, offset 0 is guaranteed to be the empty string by the spec.

File: internal/btf/types.go [hunk 2]
```diff
@@ -233,15 +233,22 @@ func (f *Func) copy() Type {
 type FuncProto struct {
 	TypeID
 	Return Type
-	// Parameters not supported yet
+	Params []FuncParam
 }
 
 func (fp *FuncProto) walk(cs *copyStack) { cs.push(&fp.Return) }
```
Comment (lmb) [inline internal/btf/types.go:239]: I think this needs to push all fp.Params.Type instances as well.

Comment (brycekahle) [reply internal/btf/types.go]: Updated.

**Review by lmb — APPROVED** [review]

**Pull request merged**